backend/src/main/java/de/jmpsoftware/backend/service/ArtNetThread.java

```diff
@@ -0,0 +1,123 @@
+    private final PipedInputStream pipe;
+    final PipeEntry pipeEntry;
+    final byte[] readByte = new byte[4];
+    long sendDMXTimeStamp;
+
+    public ArtNetThread(PipedInputStream pipe) {
+        this.pipe = pipe;
+        this.artnet = new ArtNetClient(null);
+        pipeEntry = new PipeEntry();
+        for (int i = 0; i < ArtNetService.COUNT_UNIVERSE; i++)
+            for (int ii = 0; ii < ArtNetService.SIZE_UNIVERSE; ii++)
+                dmxData[i][ii] = 0;
+
+        LOG.trace("ArtNetClient is created");
+    }
+
+    @SneakyThrows
+    @Override
+    public void run() {
+        LOG.trace("ArtNetClient Thread goes run");
+        this.artnet.start();
+        sendDMXTimeStamp = ZonedDateTime.now().toInstant().toEpochMilli();
+
+        while (true) {
+            while (pipe.available() > 0) {
+                if (checkPipeReadDoneCorrect(pipe.read(readByte, 0, 4))) {
+
+                    pipeEntry.setValuesFromBytePack(readByte);
+
+                    LOG.info("ThreadRead" + pipeEntry);
+
+                    if(isPipeEntryCommandDMX(pipeEntry)) {
+
+                        if (pipeEntry.getValuesValid())
+                            dmxData[pipeEntry.getUniverse()][pipeEntry.getChannel()] = (byte) pipeEntry.getValueDMX();
+
+                        if (!pipeEntry.isCommand(ArtNetService.COMMANDO_DMX_NO_SEND))
+                            broadcastDmx();
+                    } else {
+                        if (pipeEntry.isCommand(ArtNetService.COMMAND_NO_STOP))
+                            this.artnet.stop();
+                        else if (pipeEntry.isCommand(ArtNetService.COMMAND_NO_START))
+                            this.artnet.start();
+                    }
+                }
+            }
+            if (this.artnet.isRunning()) {
+                if (isMaxTimeBetweenLastDmxSendBigger(ARTNET_FORCE_BROADCAST_TIME)) {
+                    broadcastDmx();
+                } else {
+                    sleepThreadMillis(ACTIVE_ARTNET_DOZY_TIME);
+                }
+            } else {
+                sleepThreadMillis(SLEEPING_ARTNET_DOZY_TIME);
+            }
+        }
+    }
+
+    private void sleepThreadMillis(long millis) {
+        try {
+            Thread.sleep(millis);
+        } catch (InterruptedException e) {
+            LOG.warn("Do not Panic! But Sleep did not work.");
+        }
+    }
+
+    private boolean isMaxTimeBetweenLastDmxSendBigger(long millis) {
+        return ((ZonedDateTime.now().toInstant().toEpochMilli()) - sendDMXTimeStamp) > millis;
+    }
+
+    boolean isPipeEntryCommandDMX(PipeEntry pipeEntry) {
+        return ((pipeEntry.getCommand() >= ArtNetService.COMMAND_DMX) && (pipeEntry.getCommand() <= ArtNetService.COMMANDO_DMX_NO_SEND));
+
+    }
+
+    boolean checkPipeReadDoneCorrect(int pipeReadReturn) throws IOException {
+        if (pipeReadReturn == -1) {
+            return false;
+        }
+        if (pipeReadReturn != 4) {
```

Comment:
Maybe give names to these magic numbers? Would help me understand a little bit of what is going on ;)